### PR TITLE
ICMP type number and code number are actually "from port" and "to port"

### PR DESCRIPTION
**Pull request by phunehehe**

### CODE REVIEW

**Comment by fmap** [comment]
Can you break out the Nix changes into a separate commit?

File: nix/aws/ec2-sg.nix [hunk 1]
```diff
@@ -43,17 +43,9 @@ in
...
-            description = "ICMP type number (ICMP only, -1 for all).";
-            type = types.uniq (types.nullOr types.int);
-          };
-
-          codeNumber = mkOption {
-            default = null;
-            description = "ICMP code number (ICMP only, -1 for all).";
-            type = types.uniq (types.nullOr types.int);
-          };
+          # These are implemented in the AWS API as fromPort and toPort
```
Comment (proger) [inline nix/aws/ec2-sg.nix]: could you mention this in docs to `fromPort/toPort` instead?

File: shell.nix [hunk 2]
```diff
@@ -1,17 +1,17 @@
...
 
-  f = { mkDerivation, aeson, aeson-pretty, amazonka
-      , amazonka-autoscaling, amazonka-core, amazonka-ec2, amazonka-elb
-      , amazonka-route53, async, attoparsec, base, base64-bytestring
-      , bytestring, conduit, conduit-extra, containers, directory
-      , exceptions, filepath, lens, lifted-base, mtl, natural-sort
-      , optparse-applicative, pretty-show, process, random, resourcet
-      , scientific, semigroups, stdenv, tagged, text, time, unix
-      , unordered-containers, vector, vk-posix-pty, xml-conduit
+  f = { mkDerivation, aeson, aeson-pretty, amazonka, amazonka-core
```
Comment (proger) [inline shell.nix]: you probably want to include this in a separate PR (or remove it, as expressions are now in ug)

Comment (phunehehe) [inline shell.nix]: I need this file here to build, but OK, I'll remove and ignore it.

**Comment by fmap** [comment]
LGTM. Eventually it might be nice to have a sum-type attaching protocols to port/type/code ranges with proper bound checks. But the planner doesn't seem to have been using the earlier options at all, so this is a good intermediate step.

**Comment by phunehehe** [comment]
> Can you break out the Nix changes into a separate commit?

broken

**Comment by fmap** [comment]
Cool, let's wait for (7.8) CI.

**Pull request merged**